backend/src/main/java/vn/edu/hust/nmcnpm_20242_n3/controller/SubscriptionController.java

```diff
@@ -0,0 +1,122 @@
+package vn.edu.hust.nmcnpm_20242_n3.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.web.bind.annotation.*;
+import vn.edu.hust.nmcnpm_20242_n3.service.SubscriptionService;
+
+@RestController
+@RequestMapping("/api/subscriptions")
+public class SubscriptionController {
+
+    private final SubscriptionService subscriptionService;
+
+    @Autowired
+    public SubscriptionController(SubscriptionService subscriptionService) {
+        this.subscriptionService = subscriptionService;
+    }
+
+    // API để đăng ký nhận thông báo
+    @PostMapping("/subscribe/{userId}/{bookCopyId}")
+    public ResponseEntity<?> subscribeToBookCopy(@PathVariable int bookCopyId,@PathVariable String userId) {
+        try {
+            subscriptionService.subscribeToBookCopy(bookCopyId, userId);
+            return ResponseEntity.ok("Subscription successful");
+        } catch (Exception e) {
+            return ResponseEntity.status(500).body("Error subscribing: " + e.getMessage());
+        }
+    }
+
+    // API để hủy đăng ký nhận thông báo
+    @DeleteMapping("/unsubscribe/{userId}/{bookCopyId}")
+    public ResponseEntity<?> unsubscribeFromBookCopy(@PathVariable int bookCopyId, @PathVariable String userId) {
+        try {
+            subscriptionService.unsubscribeFromBookCopy(bookCopyId, userId);
+            return ResponseEntity.ok("Unsubscription successful");
+        } catch (Exception e) {
+            return ResponseEntity.status(500).body("Error unsubscribing: " + e.getMessage());
+        }
+    }
```

Comment:
Using the `DELETE` method here is inappropriate, as you only modify a row in the `subscription` table with the way you implemented subscriptions. Should we consider changing that?

Also, I'm pretty sure that each subscription already has an id associated with it. Should we just pass the id as input param instead of these 2?